llvm/test/Instrumentation/RealtimeSanitizer/rtsan_unsafe.ll

```diff
@@ -0,0 +1,19 @@
+; RUN: opt < %s -passes=rtsan -S | FileCheck %s
+
+define void @_Z17blocking_functionv() #0 {
+  ret void
+}
+
+define noundef i32 @main() #2 {
+  call void @_Z17blocking_functionv() #4
+  ret i32 0
+}
+
+attributes #0 = { mustprogress noinline sanitize_realtime_unsafe optnone ssp uwtable(sync) }
+
+; RealtimeSanitizer pass should create the demangled function name as a global string, and
+; pass it as input to an inserted __rtsan_expect_not_realtime call at the function entrypoint
```

Comment:
old name is in this comment.

You may consider making this comment less specific so it doesn't fall out of style as quickly "RTSan pass should insert a function call to notify the runtime of an illegal call - it passes in the demangled name of the function it is called from"

Reply:
Nice idea. Thanks